drivers/ethernet/eth_native_tap.c

```diff
@@ -307,6 +308,10 @@ static void eth_iface_init(struct net_if *iface)
 {
 	struct eth_context *ctx = net_if_get_device(iface)->data;
 	struct net_linkaddr *ll_addr = eth_get_mac(ctx);
+#if !defined(CONFIG_ETH_NATIVE_POSIX_RANDOM_MAC)
```

Comment:
```suggestion
#if !defined(CONFIG_ETH_NATIVE_TAP_RANDOM_MAC)
```
CONFIG_ETH_NATIVE_POSIX_RANDOM_MAC is deprecated in favor of CONFIG_ETH_NATIVE_TAP_RANDOM_MAC